lib/ohai/plugins/linux/network.rb

```diff
@@ -171,6 +171,36 @@ def ethernet_layer_one(iface)
     iface
   end
 
+  # determine ring parameters for the interface using ethtool
+  def ethernet_ring_parameters(iface)
+    return iface unless ethtool_binary = find_ethtool_binary
+    iface.each_key do |tmp_int|
+      next unless iface[tmp_int][:encapsulation] == "Ethernet"
+      so = shell_out("#{ethtool_binary} -g #{tmp_int}")
+      Ohai::Log.debug("Parsing ethtool output: #{so.stdout}")
+      # Nasty, brittle regex to capture the ethtool output
```

Comment:
Is there anything that would make this less nasty and brittle? Multiple queries for each match? Or just that we're reading from something that may change?